### DISCUSSION TITLE
POWERLEVEL9K_EMPTY_LINE_RIGHT_PROMPT_FIRST_SEGMENT_START_SYMBOL expected behavior?

### DISCUSSION THREAD

**Discussion by habahabahaba** [romkatv/powerlevel10k#2804] [General]
` typeset -g POWERLEVEL9K_EMPTY_LINE_RIGHT_PROMPT_FIRST_SEGMENT_START_SYMBOL='┤'`
![Screenshot 2024-12-17 at 14 29 38](https://github.com/user-attachments/assets/2efe89bf-d27d-4215-9a7d-4ca04f4b5aad)
Should the symbol at the end of a bottom line be there?

**Comment by romkatv** [comment]
There is an image at the top of https://github.com/romkatv/powerlevel10k with the examples of prompts. One of them has the full frame.

It appears that your terminal is rendering `─╮` and `─╯` incorrectly.

**Comment by habahabahaba** [comment]
Thank you for the quick response. 
I do not currently believe, that's what happening here. I fairly certain those are not ─╮ and ─╯ symbols, but a single ┤ symbol .
I didn't provide enough information. 

I do not have the right part of the frame (right bracket) turned on, only the left bracket and a filler line. I was tying to configure p10k in such a way, that it would render the ┤ symbol at the end of the filler line if there's no right segment of the prompt (like it is in my picture).

I assumed (may be erroniously), that's what `POWERLEVEL9K_EMPTY_LINE_RIGHT_PROMPT_FIRST_SEGMENT_START_SYMBOL` was for; but the  ┤ symbol, I set it equal to, is rendered at the end of both prompts (top prompt and bottom prompt). Is this the expected behavior?

Here's a bigger portion of my config:
```
  # Connect left prompt lines with these symbols.
  typeset -g POWERLEVEL9K_MULTILINE_FIRST_PROMPT_PREFIX='%8F┌'
  typeset -g POWERLEVEL9K_MULTILINE_NEWLINE_PROMPT_PREFIX='%8F├─'
  typeset -g POWERLEVEL9K_MULTILINE_LAST_PROMPT_PREFIX='%8F└─' 
  # Connect right prompt lines with these symbols.
  typeset -g POWERLEVEL9K_MULTILINE_FIRST_PROMPT_SUFFIX=
  typeset -g POWERLEVEL9K_MULTILINE_NEWLINE_PROMPT_SUFFIX=
  typeset -g POWERLEVEL9K_MULTILINE_LAST_PROMPT_SUFFIX=

  # The left end of left prompt.
  typeset -g POWERLEVEL9K_LEFT_PROMPT_FIRST_SEGMENT_START_SYMBOL=' '
  # The right end of right prompt.
  typeset -g POWERLEVEL9K_RIGHT_PROMPT_LAST_SEGMENT_END_SYMBOL=
  

  # Ruler, a.k.a. the horizontal line before each prompt. If you set it to true, you'll
  # probably want to set POWERLEVEL9K_PROMPT_ADD_NEWLINE=false above and
  # POWERLEVEL9K_MULTILINE_FIRST_PROMPT_GAP_CHAR=' ' below.
  typeset -g POWERLEVEL9K_SHOW_RULER=false
  typeset -g POWERLEVEL9K_RULER_CHAR='─'        # reasonable alternative: '·'
  typeset -g POWERLEVEL9K_RULER_FOREGROUND=8

  # Filler between left and right prompt on the first prompt line. You can set it to '·' or '─'
  # to make it easier to see the alignment between left and right prompt and to separate prompt
  # from command output. It serves the same purpose as ruler (see above) without increasing
  # the number of prompt lines. You'll probably want to set POWERLEVEL9K_SHOW_RULER=false
  # if using this. You might also like POWERLEVEL9K_PROMPT_ADD_NEWLINE=false for more compact
  # prompt.
  # typeset -g POWERLEVEL9K_MULTILINE_FIRST_PROMPT_GAP_CHAR=
  typeset -g POWERLEVEL9K_MULTILINE_FIRST_PROMPT_GAP_CHAR='─'
  if [[ $POWERLEVEL9K_MULTILINE_FIRST_PROMPT_GAP_CHAR != ' ' ]]; then
    # The color of the filler.
    typeset -g POWERLEVEL9K_MULTILINE_FIRST_PROMPT_GAP_FOREGROUND=8
    # Add a space between the end of left prompt and the filler.
    typeset -g POWERLEVEL9K_LEFT_PROMPT_LAST_SEGMENT_END_SYMBOL=' '
    # Add a space between the filler and the start of right prompt.
    typeset -g POWERLEVEL9K_RIGHT_PROMPT_FIRST_SEGMENT_START_SYMBOL='%8F┤ '
    typeset -g POWERLEVEL9K_EMPTY_LINE_LEFT_PROMPT_FIRST_SEGMENT_END_SYMBOL='%{%}'
    # End filler on the edge of the screen if there are no right segments on the first line.
    typeset -g POWERLEVEL9K_EMPTY_LINE_RIGHT_PROMPT_FIRST_SEGMENT_START_SYMBOL='┤'
  fi
```

**Comment by romkatv** [comment]
`POWERLEVEL9K_EMPTY_LINE_RIGHT_PROMPT_FIRST_SEGMENT_START_SYMBOL` is displayed on every line of prompt that has no visible prompt segments on the right hand side.

**Comment by habahabahaba** [comment]
So it is by design. Thank you!
Out of curiosity, is there a case, in which something could be ever displayed on the right side of the **bottom** prompt?

**Comment by romkatv** [comment]
The last prompt line can have prompt segments on the left and on the right, just like any other prompt line. The default Lean config does not define any segments on the right hand side of the last line, but you can change that. See https://github.com/romkatv/powerlevel10k/blob/ef83e13c22cf8641f7ab2d50cd1338d01bb31cd2/config/p10k-lean.zsh#L103-L110

**Comment by habahabahaba** [comment]
Thank you again! I'm closing this discussion now.
